### PR TITLE
Update target version to 19h1 and set XamlRoot for Flyout

### PR DESCRIPTION
**Pull request by licanhua**
Part of task #2364
To support Xaml Island, we need to set target version to 19H1 or above.

This pullrequest:
1. Target version to 10.0.18362.0
2. Workaround the cppwinrt problem by using OLD_CPPWINRT.  #2455 is create to track the cppwinrt issue
3. set XamlRoot for Flyout. Flyout doesn't have a parent. Since XamlRoot is the same in the same tree, try to get a valid XamlRoot from NativeUIManager.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/2456)

### CODE REVIEW

File: vnext/ReactUWP/Views/PopupViewManager.cpp [hunk 1]
```diff
@@ -169,6 +169,12 @@ void PopupShadowNode::SetAnchorPosition(const winrt::Popup& popup)
       auto point = popupTransform.TransformPoint(bottomRightPoint);
       popup.HorizontalOffset(point.X + popup.HorizontalOffset());
       popup.VerticalOffset(point.Y + popup.VerticalOffset());
+
+      // Set XamlRoot on the popup to handle XamlIsland/AppWindow scenarios.
+      if (auto uiElement10 = targetElement.try_as<winrt::IUIElement10>())
+      {
+        popup.XamlRoot(uiElement10.XamlRoot());
+      }
```
Comment (kmelmon) [inline vnext/ReactUWP/Views/PopupViewManager.cpp]: This bit of code is unnecessary...   Setting a XamlRoot is only required for un-parented Popups, which isn't possible, even in this case (the Popup is still parented).

File: vnext/ReactUWP/Views/FlyoutViewManager.cpp [hunk 2]
```diff
@@ -111,6 +111,18 @@ void FlyoutShadowNode::createView()
     if (!m_updating && instance != nullptr)
       OnFlyoutClosed(*instance, m_tag, false);
   });
+
+  // Set XamlRoot on the popup to handle XamlIsland/AppWindow scenarios.
```
Comment (licanhua) [inline vnext/ReactUWP/Views/FlyoutViewManager.cpp]:
```suggestion
  // Set XamlRoot on the Flyout to handle XamlIsland/AppWindow scenarios.
```

File: vnext/ReactUWP/Modules/NativeUIManager.cpp [hunk 3]
```diff
@@ -80,6 +80,25 @@ void NativeUIManager::DirtyYogaNode(int64_t tag)
   }
 }
 
+facebook::react::ShadowNode* NativeUIManager::getShadowNodeWithXamlRoot()
+{
+  if (m_host)
+  {
+    for (auto tag : m_host->GetAllRootTags())
```
Comment (StephenLPeters) [inline vnext/ReactUWP/Modules/NativeUIManager.cpp]: >auto [](start = 9, length = 4)

I'd use const auto for all these

File: vnext/ReactUWP/Views/FlyoutViewManager.cpp [hunk 4]
```diff
@@ -111,6 +111,18 @@ void FlyoutShadowNode::createView()
     if (!m_updating && instance != nullptr)
       OnFlyoutClosed(*instance, m_tag, false);
   });
+
+  // Set XamlRoot on the Flyout to handle XamlIsland/AppWindow scenarios.
+  if (auto flyoutBase6 = m_flyout.try_as<winrt::IFlyoutBase6>())
```
Comment (StephenLPeters) [inline vnext/ReactUWP/Views/FlyoutViewManager.cpp]: >auto [](start = 6, length = 4)

const

File: vnext/ReactUWP/Modules/NativeUIManager.cpp (hunk 3 above)
Comment (licanhua) [reply vnext/ReactUWP/Modules/NativeUIManager.cpp]:
```suggestion
    for (auto const tag : m_host->GetAllRootTags())
```

File: vnext/ReactUWP/Views/FlyoutViewManager.cpp (hunk 4 above)
Comment (licanhua) [reply vnext/ReactUWP/Views/FlyoutViewManager.cpp]: Is there any guideline about this? I think if (auto abc = xx) is more nature and it's widely used everywhere. and I didn't see any benefit here.

File: vnext/ReactUWP/Modules/NativeUIManager.cpp [hunk 5]
```diff
@@ -80,6 +80,25 @@ void NativeUIManager::DirtyYogaNode(int64_t tag)
   }
 }
 
+facebook::react::ShadowNode* NativeUIManager::getShadowNodeWithXamlRoot()
```
Comment (kmelmon) [inline vnext/ReactUWP/Modules/NativeUIManager.cpp]: >acebook::react::ShadowNode* [](start = 1, length = 27)

Consider changing this function to return the XamlRoot - as it stands it returns the first ShadowNode that has a XamlRoot, which makes it not very useful beyond asking it for the XamlRoot.

File: vnext/ReactUWP/Modules/NativeUIManager.h [hunk 6]
```diff
@@ -53,6 +53,12 @@ class NativeUIManager : public facebook::react::INativeUIManager
   // Other public functions
   void DirtyYogaNode(int64_t tag);
 
+  // For unparented node like Flyout, XamlRoot should be set to handle XamlIsland/AppWindow scenarios.
+  // Since it doesn't have parent, and all nodes in the tree should have the same XamlRoot,
+  // this function iterates all roots and try to get a valid XamlRoot.
+  // To reduce the dependency to cppwinrt in header file, ShadowNode* other XamlRoot is returned.
```
Comment (kmelmon) [inline vnext/ReactUWP/Modules/NativeUIManager.h]: >ShadowNode* other XamlRoot is returned [](start = 58, length = 38)

Can you forward declare XamlRoot to solve this?

Comment (licanhua) [reply vnext/ReactUWP/Modules/NativeUIManager.h]: I just changed it to winrt::XamlRoot.

File: vnext/ReactUWP/Modules/NativeUIManager.cpp (hunk 5 above)
Comment (licanhua) [reply vnext/ReactUWP/Modules/NativeUIManager.cpp]: Done

**Pull request merged**